Json/Data.ts

```diff
@@ -0,0 +1,36 @@
+import { isly } from "isly"
+
+export type Data = boolean | string | number | null | undefined | Data.Object | Data.Array
```

Comment:
There seams to be something wrong with this type. 
With This code:
```ts
interface User {
	name: { first: string; last: string }
	roles: string[]
}

function foo<T extends Data>(data: T): T {
	return data
}

foo<User>({ name: { first: "Jessie", last: "Doe" }, roles: [] })
```
There is a typescript error: 
```
Type 'User' does not satisfy the constraint 'Data'.
  Type 'User' is not assignable to type 'Object'.
    Index signature for type 'string' is missing in type 'User'
```

Reply:
Fixed in rewrite.